src/main/groovy/com/rundeck/plugins/ansible/ansible/AnsibleVault.java

```diff
@@ -71,58 +84,122 @@ public String encryptVariable(String key,
-
-            final InputStream stdoutInputStream = proc.getInputStream();
-            final BufferedReader stdoutReader = new BufferedReader(new InputStreamReader(stdoutInputStream));
-
-            String line1 = null;
-            boolean capture = false;
-            while ((line1 = stdoutReader.readLine()) != null) {
-                if (line1.toLowerCase().contains("!vault")) {
-                    capture = true;
-                }
-                if (capture) {
-                    stringBuilder.append(line1).append("\n");
+            final InputStream proccesInputStream = proc.getInputStream();
+            final OutputStream processOutputStream = proc.getOutputStream();
+
+            //capture output thread
+            Callable<String> readOutputTask = () -> {
+                return readOutput(proccesInputStream);
+            };
+
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Future<String> future = executor.submit(readOutputTask);
+
+            Thread stdinThread = new Thread(() -> sendValuesStdin(processOutputStream, masterPassword, content));
+
+            long start = System.currentTimeMillis();
+            long end = start + 60 * 1000;
+
+            //wait for prompt
+            boolean promptFound = false;
+            while (!promptFound && System.currentTimeMillis() < end) {
+                BufferedReader reader = new BufferedReader(new FileReader(promptFile));
+                String currentLine = reader.readLine();
+                if(currentLine!=null && currentLine.contains("Enter Password:")){
+                    promptFound = true;
+                    //send password / content
+                    stdinThread.start();
+                    reader.close();
+                }else{
+                    Thread.sleep(1500);
                 }
             }
 
+            if(!promptFound){
+                throw new RuntimeException("Failed to find prompt for ansible-vault");
+            }
+
             int exitCode = proc.waitFor();
 
+            //get encrypted value
+            String result = future.get();
+            executor.shutdown();
+
             if (exitCode != 0) {
-                System.err.println("ERROR: encryptFileAnsibleVault:" + procArgs);
-                return null;
+                throw new RuntimeException("ERROR: encryptFileAnsibleVault:" + procArgs);
             }
-            return stringBuilder.toString();
+            return result;
 
         } catch (Exception e) {
-            System.err.println("error encryptFileAnsibleVault file " + e.getMessage());
-            return null;
+            throw new RuntimeException("Failed to encrypt variable: " + e.getMessage());
         } finally {
             // Make sure to always cleanup on failure and success
             if (proc != null) {
                 proc.destroy();
             }
+
+            if(promptFile!=null && promptFile.delete()){
```

Comment:
iirc `delete()` returns true on success. Should this be `!promptFile.delete()` ?

Reply:
I wonder if the delete on exit is helpful here. If it didn't delete successfully here, would we expect the permissions to be different when the JVM terminates?

Reply:
right